apps/browser-extension-wallet/src/providers/FeatureFlags/context.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { createContext, useContext, useEffect, useState } from 'react';
+import { useAnalyticsContext } from '@providers/AnalyticsProvider';
+
+interface FeatureFlagsProviderProps {
+  children: React.ReactNode;
+}
+
+type FeatureFlags = {
+  flagsHaveBeenLoaded: boolean;
+};
+
+// eslint-disable-next-line unicorn/no-null
+const FeatureFlagsContext = createContext<FeatureFlags | null>(null);
+
+export const useFeatureFlagsContext = (): FeatureFlags => {
+  const analyticsContext = useContext(FeatureFlagsContext);
+  if (analyticsContext === null) throw new Error('context not defined');
+  return analyticsContext;
+};
+
+// eslint-disable-next-line arrow-body-style
+export const FeatureFlagsProvider = ({ children }: FeatureFlagsProviderProps): React.ReactElement => {
+  const analytics = useAnalyticsContext();
+  const [flagsHaveBeenLoaded, setFlagsHaveBeenLoaded] = useState(false);
```

Comment:
How about using this shorter form?
```suggestion
  const [flagsLoaded, setFlagsLoaded] = useState(false);
```